src/pveclib/vec_f128_ppc.h

```diff
@@ -951,17 +1314,352 @@ test_cmpltf128_v3d (vui128_t vfa128, vui128_t vfb128)
+#elif  defined (_ARCH_PWR8)
+  vui64_t q_exp;
+  vui128_t q_sig;
+  const vui128_t q_zero = (vui128_t) { 0 };
+  const vui32_t lowmask = CONST_VINT128_W( 0, 0, 0, 1);
+
+  q_sig = int128;
+  // Quick test for 0UL as this case requires a special exponent.
+  if (vec_cmpuq_all_eq (q_sig, q_zero))
+    {
+      result = vec_xfer_vui128t_2_bin128 (q_zero);
+    }
+  else
+    { // We need to produce a normal QP, so we treat the QW integer
+      // like a denormal, then normalize it.
+      // Start with the quad exponent bias + 127 then subtract the count of
+      // leading '0's. The 128-bit sig can have 0-127 leading '0's.
+      vui64_t q_expm = (vui64_t) CONST_VINT64_DW(0, (0x3fff + 127));
+      vui64_t i64_clz = (vui64_t) vec_clzq (q_sig);
+      q_sig = vec_slq (q_sig, (vui128_t) i64_clz);
+      q_exp = vec_subudm (q_expm, i64_clz);
+      // This is the part that might require rounding.
+      // For example Round to zero
+      // Shift right 15-bits to normalize and truncate
+      q_sig = vec_srqi ((vui128_t) q_sig, 15);
+      //...
+      q_exp = vec_swapd (q_exp);
+      result = vec_xsiexpqp (q_sig, q_exp);
+    }
+ * \endcode
+ * In this example the Significand (including the L-bit) is right
+ * justified in the high-order 113-bits of q_sig.
+ * The guard, round, and sticky (GRX) bits are in the low-order
+ * 15 bits.
+ * The sticky-bits are the last 13 bits and are logically ORed
+ * (or added to 0x1fff) to produce the X-bit.
+ *
+ * The signed quadword conversion is bit more complicated for both
+ * POWER9/8. For example:
+ * \code
+__binary128
+static inline vec_xscvsqqp (vi128_t int128)
+{
+  __binary128 result;
+#if defined (_ARCH_PWR10)  && (__GNUC__ >= 10)
+  __asm__(
+      "xscvsqqp %0,%1"
+      : "=v" (result)
+      : "v" (int128)
+      : );
+#elif defined (_ARCH_PWR9) && defined (__FLOAT128__) && (__GNUC__ > 7)
+  __binary128 hi64, lo64, i_sign;
+  __binary128 two64 = 0x1.0p64;
+  vui128_t q_sig;
+  vui32_t q_sign;
+  vui128_t q_neg;
+  vb128_t b_sign;
+  const vui32_t signmask = CONST_VINT128_W(0x80000000, 0, 0, 0);
+  // Collect the sign bit of the input value.
+  q_sign = vec_and ((vui32_t) int128, signmask);
+  // Convert 2s complement to unsigned magnitude form.
+  q_neg  = (vui128_t) vec_negsq (int128);
+  b_sign = vec_setb_sq (int128);
+  q_sig = vec_seluq ((vui128_t) int128, q_neg, b_sign);
+  // generate a signed 0.0 to use with vec_copysignf128
+  i_sign = vec_xfer_vui32t_2_bin128 (q_sign);
+  // Convert the unsigned int128 magnitude to __binary128
+  vui64_t int64 = (vui64_t) q_sig;
+  hi64 = int64[VEC_DW_H];
+  lo64 = int64[VEC_DW_L];
+  result = (hi64 * two64) + lo64;
+  // copy the __int128's sign into the __binary128 result
+  result = vec_copysignf128 (result, i_sign);
+#elif  defined (_ARCH_PWR8)
+...
+#endif
+  return result;
+}
+ * \endcode
+ * For POWER9
+ * we can not just used the signed doubleword conversions for this case.
+ * First we convert the signed quadword into a sign bool and unsigned
```

Comment:
`sign bool` is awkward, but OK. `sign (bool)`, perhaps?

Reply:
How about ...
First we split the signed quadword into a 128-bit boolean
(representing the sign) and an unsigned quadword magnitude.